### PR TITLE
:monocle_face: Add parsing and error handling for the file

### LINKED ISSUE #9: :monocle_face: Parse the given file
# :white_check_mark: What's needed
You must parse the file given in the `-f` option and return an appropriate data type.
Here's an exemple of a given file:

image

To parse it, there's some steps detailed below.

## 📋 Subtasks
- [ ] Read the list of pixels from the file passed as argument, according to the following grammar (x,y) (r,g,b) *(See an exemple above)*
- [ ] Return `Conf` data type which contains list of vectors. Each vector contains its coordinates and its color.
- [ ] Handle the errors from the given file.

## :file_folder: Files required
- [ ] `FileParsing.hs`

### PR DESCRIPTION
**Pull request by Thomaltarix**
I added the parsing and the error handling of the given file.
---
The parsing consists into transforming each line of the file into a `tuple` of `Vectors`.
A `Vector` for the coordinates and a `Vector` for the colors.
*Each `Vector` contains a list of `Double`.*

---
The errors that have to be managed are:
- A negative value *(each value must be positive)*
- A `non-short` color number *(each color value must be >=0 and <= 255)*
- Different vector sizes *(each vector must have the same number of coordinates and the same number of colors)*

### CODE REVIEW

File: src/ImageCompressor.hs [hunk 1]
```diff
@@ -11,5 +11,6 @@ import FileParsing (Conf(..))
 import Parsing (Options(..))
 
 imageCompressor :: Maybe Conf -> Maybe Options -> IO ()
-imageCompressor (Just _) (Just _) = putStrLn "imageCompressor no error"
-imageCompressor _ _ = putStrLn "imageCompressor error"
+imageCompressor (Just _) (Just _) = putStrLn "Success"
+imageCompressor Nothing _ = putStrLn "Wrong file content"
+imageCompressor _ _ = putStrLn "Wrong Options"
```
Comment (landryarki) [inline src/ImageCompressor.hs:16]: isn't there a way to specify which options contain an error (without getting too complicated) ?

**Review by landryarki — CHANGES_REQUESTED** [review]
not necessarily a request for change, more an insistent comment

File: src/ImageCompressor.hs (hunk 1 above)
Comment (Thomaltarix) [reply src/ImageCompressor.hs:16]: I didn't figured out how to properly catch errors yet. At the moment the pull-request just miss some error handling. But my goal is to push the main functionnalities to make you able to test your algorithm while developping it. Then, on my side, i'll create a new issue to improve the existing error handling 👍

Comment (Thomaltarix) [reply src/ImageCompressor.hs:16]: I created this issue #11 to match what i said.

**Review by landryarki — APPROVED** [review]
:+1:

File: src/ImageCompressor.hs (hunk 1 above)
Comment (landryarki) [reply src/ImageCompressor.hs:16]: good

**Pull request merged**